luarules/gadgets/unit_prevent_self_guard.lua

```diff
@@ -0,0 +1,32 @@
+function gadget:GetInfo()
+    return {
+        name      = "Prevent Self Guard",
+        desc      = "Prevents all units from issuing guard commands on themselves",
+        author    = "Trunks",
+        date      = "2025",
+        enabled   = true,
+        layer     = 0,
+		license = "GNU GPL, v2 or later",
+    }
+end
+
+if not gadgetHandler:IsSyncedCode() then
+    return
+end
+
+local CMD_GUARD = CMD.GUARD
+
+function gadget:Initialize()
+    gadgetHandler:RegisterAllowCommand(CMD_GUARD)
```

Comment:
no need to localize a var that is only ever used once.
```suggestion
function gadget:Initialize()
    gadgetHandler:RegisterAllowCommand(CMD.GUARD)
```